pkg/customresourcestate/config.go

```diff
@@ -158,21 +166,47 @@ type ConfigDecoder interface {
 	Decode(v interface{}) (err error)
 }
 
-// FromConfig decodes a configuration source into a slice of customresource.RegistryFactory that are ready to use.
-func FromConfig(decoder ConfigDecoder) ([]customresource.RegistryFactory, error) {
-	var crconfig Metrics
+// FromConfig decodes a configuration source into a slice of `customresource.RegistryFactory` that are ready to use.
+func FromConfig(decoder ConfigDecoder, r *discovery.GVKMaps) ([]customresource.RegistryFactory, error) {
+	var customResourceConfig Metrics
 	var factories []customresource.RegistryFactory
 	factoriesIndex := map[string]bool{}
-	if err := decoder.Decode(&crconfig); err != nil {
+	if err := decoder.Decode(&customResourceConfig); err != nil {
 		return nil, fmt.Errorf("failed to parse Custom Resource State metrics: %w", err)
 	}
-	for _, resource := range crconfig.Spec.Resources {
+	// Cannot proceed ahead without an instantiated GVKMaps object (only for the
+	// first time, otherwise the cache will come up empty on the first run (in the
+	// <30s timespan).
+	for {
+		if !r.Heartbeat.IsZero() {
+			break
+		}
+	}
+	type dGVK = discovery.GroupVersionKind
+	var resources []Resource
+	for _, resource := range customResourceConfig.Spec.Resources {
+		gotGVKs := r.ResolveGVK(dGVK(resource.GroupVersionKind))
```

Comment:
I would prefer not to use a new type here.

```suggestion
	var resources []Resource
	for _, resource := range customResourceConfig.Spec.Resources {
		gotGVKs := r.ResolveGVK(discovery.GroupVersionKind(resource.GroupVersionKind))
```